File: src/SharpOMatic.Engine/Repository/Setting.cs
namespace SharpOMatic.Engine.Repository;

public class Setting
{
    [Key]
    public required Guid SettingId { get; set; }
    public required string Name { get; set; }
    public required SettingType SettingType { get; set; }
    public string? ValueString { get; set; }
    public bool? ValueBoolean { get; set; }
    public int? ValueInteger { get; set; }
    public double? ValueDouble { get; set; }
}

File: src/SharpOMatic.Editor/src/app/pages/settings/interfaces/setting.ts
import { SettingType } from '../../../enumerations/setting-type';

export interface Setting {
  settingId: string;
  name: string;
  displayName: string;
  settingType: SettingType;
  valueString?: string | null;
  valueBoolean?: boolean | null;
  valueInteger?: number | null;
  valueDouble?: number | null;
}
